main/views.py

```diff
@@ -1,57 +1,198 @@
+            winner_id = match.player2.id
+            loser_name = match.player1.name
+            loser_id = match.player1.id
+            winner_score = match.player2_score
+            loser_score = match.player1_score
+        results.append({
+            "id": match.id,
+            "date": match.date_and_time,
+            "winner": {
+                "id": winner_id,
+                "name": winner_name,
+                "score": winner_score,
+            },
+            "loser": {
+                "id": loser_id,
+                "name": loser_name,
+                "score": loser_score,
+            },
+            "field": {
+                "id": match.field.id,
+                "name": match.field.name,
+            }
+        })
     context = {
-        'matches': matches,
         'results': results,
     }
     return render(request, 'main/index.html', context)
 
+
 def players(request):
     players = Player.objects.all()
     context = {
         'players': players,
     }
     return render(request, 'main/players.html', context)
 
+
 def fields(request):
     fields = Field.objects.all()
     context = {
         'fields': fields,
     }
     return render(request, 'main/fields.html', context)
 
+
 def player_detail(request, player_id):
     player = get_object_or_404(Player, pk=player_id)
     return render(request, 'main/player_detail.html', {'player': player})
 
+
 def api_players(request):
-    players = Player.objects.all()
-    data = []
-    for player in players:
-        data.append({"id": player.id, "name": player.name})
-    return JsonResponse({"players": data})
+    if request.method == "GET":
+        players = Player.objects.all()
+        data = [{"id": player.id, "name": player.name} for player in players]  # should I be doing the hanging indents here?
+        return JsonResponse({"players": data})
+    elif request.method == "POST":
+        data = json.loads(request.body)
+        player = Player.objects.create(name=data["name"])
+        return JsonResponse({
+            "id": player.id,
+            "name": player.name
+        })
+    else:
+        return HttpResponseNotAllowed(["POST", "GET"])
+
+
+def api_fields(request):
+    if request.method == "GET":
+        fields = Field.objects.all()
+        data = [{"id": field.id, "name": field.name} for field in fields]
+        return JsonResponse({
+            "fields": data
+        })
+    elif request.method == "POST":
+        data = json.loads(request.body)
+        field = Field.objects.create(name=data["name"])
+        return JsonResponse({
+            "id": field.id,
+            "name": field.name
+        })
+    else:
+        return HttpResponseNotAllowed(["POST", "GET"])
+
+
+def api_matches(request):
+    if request.method == "GET":
+        matches = Match.objects.all()
+        # where should I put the curly brackets here based on usual BIO conventions, generally should this be done this way? could be hard to read.
```

Comment:
I'd either go with
```Python
data = [
	{
	    "id": match.id,
	    "date_and_time": match.date_and_time,
	    "player1": match.player1.name,
	    "player2": match.player2.name,
	    "player1_score": match.player1_score,
	    "player2_score": match.player2_score,
	    "field": match.field.name
	} for match in matches
]
```

or realise that this expression will be hard to read no matter how I play with the indentation, and just define a small helper function like
```Python
def get_match_details(match):
	return {
	    "id": match.id,
	    "date_and_time": match.date_and_time,
	    "player1": match.player1.name,
	    "player2": match.player2.name,
	    "player1_score": match.player1_score,
	    "player2_score": match.player2_score,
	    "field": match.field.name
	}
```
and then I can just do
```Python
data = [get_match_details(match) for match in matches]
```
which is nice and easy to read.